### PR TITLE
feat(test): rename --allow-none to --permit-no-files

### LINKED ISSUE #24752: Rename confusing --allow-none flag in `deno test`
The `deno test` command currently includes an `--allow-none` flag, which prevents returning an error code if no test files are found. This flag name is misleading as it does not align with other `--allow-*` flags related to permissions and security.

Consider renaming this flag to a more intuitive and clear alternative that better reflects its function.

Possible name alternatives:
* `--no-error-on-missing` 
* `--proceed-if-none`

### PR DESCRIPTION
**Pull request by lucacasonato**
Fixes #24752

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -4436,7 +4445,17 @@ fn test_parse(flags: &mut Flags, matches: &mut ArgMatches) {
...
   }
   let doc = matches.get_flag("doc");
-  let allow_none = matches.get_flag("allow-none");
+  #[allow(clippy::print_stderr)]
+  let allow_none = matches.get_flag("permit-no-files")
+    || if matches.get_flag("allow-none") {
+      eprintln!(
+        "⚠️ {}",
+        crate::colors::yellow("The `--allow-none` flag is deprecated and will be removed in Deno 2.0.\nUse the `--permit-no-files` flag instead."),
+      );
```
Comment (dsherret) [inline cli/args/flags.rs:4454]: Can you add a test for all this? Is `--allow-none` even used in a test?

Comment (lucacasonato) [reply cli/args/flags.rs:4454]: It seems only in a flags test. I'll add a spec test

**Review by dsherret — APPROVED** [review]
LGTM
